src/pages/NFTs/Profile/Card.tsx

```diff
@@ -1,72 +1,124 @@
+    if (connected) {
+      await fetchExternalNFTs(publicKey, connection, data)
+      setModalOpen(true)
+    }
+  }
+
   return (
-    <div className="card-item">
-      <StyledCard $border={border}>
-        <img className="card-image" src={data.image} alt="" onClick={goMyCreatedNFT} />
-        <div className="info">
-          <div className="name">{data.name}</div>
-          <div className="number">
-            <img className="check-icon" src={`/img/assets/check-icon.png`} alt="" />
-          </div>
-          <div className="other">price</div>
-          {data.name === 'favorited' ? (
-            <div className="like-group favorited-group">
-              <img className="heart-red" src={`/img/assets/heart-red.svg`} alt="" />
-              <span className="like-count">{data.name}</span>
-            </div>
-          ) : (
-            <div className="like-group">
-              <>
-                <img className="heart-purple" src={`/img/assets/heart-purple.svg`} alt="" />
-                <img className="heart-empty" src={`/img/assets/heart-empty.svg`} alt="" />
-              </>
-              <span className="like-count">0</span>
+    <>
+      {modalOpen && (
+        <STYLED_MODAL>
+          <NFTDetails mode="my-external-NFT" arbData={data} handleClickPrimaryButton={handleClickPrimaryButton} />
+        </STYLED_MODAL>
+      )}
+      <div className="card-item" onClick={() => openModal()}>
+        <StyledCard $border={border}>
+          <img className="card-image" src={data.image} alt="" onClick={goMyCreatedNFT} />
+          <div className="info">
+            <div className="name">{data.name}</div>
+            <div className="number">
+              {data.collection && <img className="check-icon" src={`/img/assets/check-icon.png`} alt="" />}
             </div>
-          )}
-          <div className="option">
-            {isExplore ? (
-              <button className="buy-now-btn">Buy now</button>
-            ) : data.name === 'favorited' ? (
-              <div className="price-group">
-                <span className="text">Last</span>
-                <img className="price-image" src={`/img/assets/price.svg`} alt="" />
-                <span className="price-number">35</span>
+
+            {data.name === 'favorited' ? (
+              <div className="like-group favorited-group">
+                <img className="heart-red" src={`/img/assets/heart-red.svg`} alt="" />
+                <span className="like-count">{data.name}</span>
               </div>
-            ) : [1, 2].includes(data.name.length) ? (
-              <button className="sell-now-btn" onClick={() => setVisible(true)}>
-                Sell now
-              </button>
             ) : (
-              <img className="card-logo" src={`/img/assets/card-logo-1.svg`} alt="" />
+              <div className="like-group">
+                <>
+                  {/* <img className="heart-purple" src={`/img/assets/heart-purple.svg`} alt="" /> */}
```

Comment:
I had no way to get the donate data from either external or internal source so I  commented out the icon for now